### DISCUSSION TITLE
Add background images on Yolov5 dataset

### DISCUSSION THREAD

**Discussion by unrue** [ultralytics/yolov5#12681] [General]
Dear yolov5 users/developers,

I'm training a model on 30k images and 75 classes (not from COCO).  Now, I would like to add background images, I read many previous discussion but I still don't understand how to download  or generate and add to my dataset. Such background images:

1) Shoud be black images?
2) Shoud be random images taken somewhere?
3) I have to add "\__background__\" class in yaml dataset configuration file, If I understand well ?
4) I have to add anntation for each background image?
5) In the confusion matrix, background class is already present, so I have to do nothing more?

I read such guide:

[https://y-t-g.github.io/tutorials/bg-images-for-yolo/](url)

And It generates me 1000 of background images (?). They seems to be random images but I don't understand which is the criteria. In such images thre are present "people" and "car" too. I expected to have images with no person and car, according to the example. However, I have 75 classes, in my dataset (person is present, car not), So?

Someone could explain me step-by-step what I have to do? Thanks.

**Comment by unrue** [comment]
@glenn-jocher could you help me? Thanks.

**Comment by Y-T-G** [comment]
1. No
2. Yes, but shouldn't include the classes you're detecting
3. No
4. No. Empty annotation files only.
5. You don't have to anything for that.

I updated the guide with a fix. It shouldn't download images with person or car now. Follow the guide again.

**Comment by unrue** [comment]
Thanks for the clarification.
About point 4, so I have to generate an empty txt file for each background image? I suppose into training set.

**Reply by Y-T-G** [reply]
Yes. But if you follow the guide, it should have generated that automatically. Just need to move them to the labels folder.

**Comment by unrue** [comment]
Thanks,

just last question: having 19.000 images on training set, how bg images do you suggest to add?

**Reply by Y-T-G** [reply]
1k should be enough
